clang/lib/AST/Type.cpp

```diff
@@ -2651,6 +2651,8 @@ bool QualType::isTriviallyRelocatableType(const ASTContext &Context) const {
     return false;
   } else if (!BaseElementType->isObjectType()) {
     return false;
+  } else if (BaseElementType.isTriviallyCopyableType(Context)) {
+    return true;
```

Comment:
Shouldn't this be after the canPassInRegisters test? at the very least it would be more efficient

Reply:
OK! I changed the order.